pkg/components/custom/registry_test.go

```diff
@@ -0,0 +1,72 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package custom
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	grpc_go "google.golang.org/grpc"
+)
+
+//custom mock object
+type mockcustom struct {
+	mock.Mock
```

Comment:
Most Go people try to avoid using mocking libraries.  Can we simply put the interface here instead and have fields for each method call response?